llama-index-integrations/embeddings/llama-index-embeddings-llamafile/llama_index/embeddings/llamafile/base.py

```diff
@@ -0,0 +1,94 @@
+from typing import Optional
+
+import httpx
+from httpx import Timeout
+
+from llama_index.core.base.embeddings.base import BaseEmbedding, Embedding
+from llama_index.core.bridge.pydantic import Field
+from llama_index.core.callbacks.base import CallbackManager
+
+DEFAULT_REQUEST_TIMEOUT = 30.0
+
+
+class LlamafileEmbedding(BaseEmbedding):
```

Comment:
Do llama file models support batch embeddings? If so, we can implement `_get_text_embeddings()` and `_aget_text_embeddings()`